lighthouse-core/index.js

```diff
@@ -30,16 +29,17 @@ const Config = require('./config/config');
  * @param {string} url
  * @param {LH.Flags} flags
  * @param {LH.Config.Json|undefined} configJSON
- * @return {Promise<LH.RunnerResult>}
+ * @return {Promise<LH.RunnerResult|undefined>}
  */
-function lighthouse(url, flags = {}, configJSON) {
```

Comment:
this is a breaking change for `lighthouse('http://example.com')` in node, do we really need to do this?

Reply:
do we support fully optional `flags`? I hadn't dug into it so I didn't pursue it (`{}` isn't currently castable to `LH.Flags` which is why I dropped it), but if so, agreed we should support that here.

I can add a manual cast of a `{}` here, maybe, and then follow up with solidifying the `flags` pipeline in a follow up?

Reply:
would `Partial<LH.Flags>` cut it? I guess we could differentiate LH.Flags from LH.CLIFlags which is basically what's defined now

Reply:
> would Partial<LH.Flags> cut it? I guess we could differentiate LH.Flags from LH.CLIFlags which is basically what's defined now

yeah, I think that's basically what we need to do. Handy to have output from yargs, but then our optional flags fed into Config is a different beast (and it looks like completely optional is fine).

Ok if we do a future "Make Flags Good" PR while leaving this with the `{}` default?

Reply:
👍